### ISSUE TITLE
does script tag templates really work?

### ISSUE THREAD

**Issue by PEZ** [juvenn/mustache.vim#5]
I don't get mustache code inside `<script type="text/mustache"...` syntax colored. The example.mustache file highlights correctly.

**Comment by amadeus** [comment]
Any update on support for this?

**Comment by bruno-** [comment]
Hi guys,
I have looked into this a bit.

**TL;DR**
In order for this to work properly `html.vim` syntax file should be updated. This can't be done through `mustache.vim` plugin.
There is a hack to make it work. Put this in your `.vimrc` file:

``` viml
if has("autocmd")
  au FileType html setlocal syntax=mustache
endif
```

**Full explanation:**
The reason why `<script type="text/mustache"` does not highlight correctly is because you are (most likely) working with plain html files (files with extension `.html` ie. `index.html`). If that is the case, then `mustache.vim` plugin does not even get loaded and mustache syntax highlighting is naturally not working.

A slight off-topic to clarify: vim determines filetype (and applies syntax highlighting) by looking at file extension.
If file extension is something of the following: `*.mustache,*.handlebars,*.hbs,*.hogan,*.hulk,*.hjs`, then proper mustache syntax highlighting will be applied.
If file extension is `.html,.htm` - in that case vim applies only basic html syntax highlighting.

I hope that explains why script tag with mustache type attribute does not "work correctly".

It also indicates another thing - if we want it to work correctly, we would have to extend plain html syntax file that comes with vim (`mustache.vim` plugin cannot influence html syntax files).
Here is [the guy that maintains `html.vim` syntax file](http://www.fleiner.com/vim/download.html), and here is the [source of the syntax file itself](http://www.fleiner.com/vim/syntax_60/html.vim).

I don't even know if any kind of pull request is accepted for `html.vim`, how the process goes, would the maintainer allow that ("text/mustache" does not look like proper mime type - officially supported as html script type attribute value). Long story short, I think this option is off.

So with proper options off, here is the hack how to get the desired result. In your `.vimrc` file put the following:

``` viml
if has("autocmd")
  au FileType html setlocal syntax=mustache
endif
```

The above command will apply `mustache` syntax highlighting for **all** `html` files. I've tested and this works just fine. Please let me know if it works for you.

**Comment by bruno-** [comment]
I think we gave it enough time for discussion. Closing the issue based on detailed explanation in previous post.
